magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -56,16 +56,25 @@ object ASG extends AWS {
     client.updateAutoScalingGroup(
       new UpdateAutoScalingGroupRequest().withAutoScalingGroupName(name).withMaxSize(capacity))
 
-  def isStabilized(asg: AutoScalingGroup, asgClient: AmazonAutoScalingClient, elbClient: AmazonElasticLoadBalancingClient) = {
+  def isStabilized(asg: AutoScalingGroup, asgClient: AmazonAutoScalingClient, elbClient: AmazonElasticLoadBalancingClient): Either[String, Boolean] = {
     elbName(asg) match {
       case Some(name) => {
         val elbHealth = ELB.instanceHealth(name, elbClient)
-        elbHealth.size == asg.getDesiredCapacity && elbHealth.forall( instance => instance.getState == "InService")
+        if (elbHealth.size != asg.getDesiredCapacity)
+          Left(s"Number of ELB instances (${elbHealth.size}) and ASG desired capacity (${asg.getDesiredCapacity}) don't match")
+        else if (!elbHealth.forall( instance => instance.getState == "InService"))
+          Left(s"Only ${elbHealth.count(_.getState == "InService")} of ${elbHealth.size} ELB instances InService")
+        else
+          Right(true)
```

Comment:
if it's only ever Right(true) then why not make it Right(())  i.e. unit? otherwise it's confusing.

The other option is to make it an Option[String] where it's an optional error.  But that's going to seriously mess with people's minds!